sfu/canvas-lms
# Prevent course duplication due to minor letter case differences

*This fixes tickets like INC0128080.*

This includes two main changes to prevent duplication of courses with letters in the course number, e.g. ECON 355W and FAL X99.

##### 1. Course identifiers are now given in lower case

Course data returned by `AmaintController#courses_for_user` contains two identifiers. The `sis_source_id` is used by the Start a New Course form to compare with other courses and prevent duplication. The `key` is used by the CSV builder to create courses.

This essentially takes case-sensitivity out of the comparison equation for credit courses.

##### 2. Course codes are now shown in upper case

Several parameters returned by `CourseDataController#search` used to be in lower case. This change is cosmetic in nature, and makes Start a New Course form search results match those of the Academic Calendar and Course Outlines.

---

Test plan:

* Use/find a test user who'll teach in a course with letters in the code
* Go to the Start a New Course form on behalf of the user
* Search for the same course that is listed in Suggested Courses
* Verify the form warns you that the course is already listed
* Select that course in Suggested Courses and create it
* Repeat these steps to search for the same course again
* Verify the form warns you that the course already exists in Canvas
